### PR TITLE
XLite support

### PR DESCRIPTION
**Pull request by bsongis**
[X10] Serial on external module
[X10] Many fixes
[ARM] Backlight check moved to readKeysAndTrims() on all ARM boards. So that the backlight is again ON when pressing a trim or the SHIFT key

### CODE REVIEW

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -285,8 +285,13 @@ void generalDefault()
...
   g_eeGeneral.vBatWarn = 66;
-  g_eeGeneral.vBatMin = -28; // 6.2V
-  g_eeGeneral.vBatMax = -38;   // 8.2V
+  g_eeGeneral.vBatMin = -28;  // 6.2V
+  g_eeGeneral.vBatMax = -38;  // 8.2V
+#elif defined(PCBXLITE)
+  // 2 x Li-Ion
+  g_eeGeneral.vBatWarn = 66;
+  g_eeGeneral.vBatMin = -23;  // 6.7V
+  g_eeGeneral.vBatMax = -37;  // 8.3V
```
Comment (kilrah) [inline radio/src/opentx.cpp]: Hmm I'd change for X10 too, it's supposed to be the same after all...

File: radio/src/keys.cpp [hunk 2]
```diff
@@ -106,7 +107,10 @@ void Key::input(bool val)
       if (m_cnt == KEY_LONG_DELAY) {
         // generate long key press
         // TRACE("key %d LONG", key());
-        putEvent(EVT_KEY_LONG(key()));
+#if !defined(LUA)
```
Comment (bsongis) [inline radio/src/keys.cpp]: It shouldn't be dependent of LUA / NO LUA

File: radio/src/keys.cpp [hunk 3]
```diff
@@ -96,7 +96,8 @@ void Key::input(bool val)
       break;
     case KSTATE_START:
       // TRACE("key %d FIRST", key());
-      putEvent(EVT_KEY_FIRST(key()));
+      if (!IS_SHIFT_KEY(key()))
```
Comment (bsongis) [inline radio/src/keys.cpp]: Perhaps this test can be removed, I think the only problem we have with the SHIFT key is KEY_REPEAT

**Comment by kilrah** [comment]
Current status:
- Xlite reboots fine from bootloader exit and a watchdog reset.
- Xlite recovery from USB mass storage is still random.
- X9E does not reboot from either bootloader exit or watchdog reset, but that is the case on the 2.2 branch as well so it got broken earlier.
- Page key and trims are broken on X7.

File: radio/src/gui/128x64/model_setup.cpp [hunk 4]
```diff
@@ -700,7 +729,7 @@ void menuModelSetup(event_t event)
 #endif
         break;
       }
-#if defined(PCBTARANIS)
+  #if defined(PCBTARANIS)
```
Comment (bsongis) [inline radio/src/gui/128x64/model_setup.cpp]: indentation

File: radio/src/gui/128x64/navigation.cpp [hunk 5]
```diff
@@ -344,7 +346,7 @@ int checkIncDec(event_t event, int val, int i_min, int i_max, unsigned int i_fla
 #if defined(PCBXLITE)
   if (s_editMode > 0) {
 #endif
-    if (event==EVT_KEY_FIRST(KEY_RIGHT) || event==EVT_KEY_REPT(KEY_RIGHT) || (s_editMode>0 && (IS_ROTARY_RIGHT(event) || event==EVT_KEY_FIRST(KEY_UP) || event==EVT_KEY_REPT(KEY_UP)))) {
+    if (event==EVT_KEY_FIRST(KEY_RIGHT) || event==EVT_KEY_REPT(KEY_RIGHT)) {
```
Comment (bsongis) [inline radio/src/gui/128x64/navigation.cpp]: you break SKY9X here

File: radio/src/gui/128x64/navigation.cpp [hunk 6]
```diff
@@ -355,7 +357,7 @@ int checkIncDec(event_t event, int val, int i_min, int i_max, unsigned int i_fla
         AUDIO_KEY_ERROR();
       }
     }
-    else if (event==EVT_KEY_FIRST(KEY_LEFT) || event==EVT_KEY_REPT(KEY_LEFT) || (s_editMode>0 && (IS_ROTARY_LEFT(event) || event==EVT_KEY_FIRST(KEY_DOWN) || event==EVT_KEY_REPT(KEY_DOWN)))) {
+    else if (event==EVT_KEY_FIRST(KEY_LEFT) || event==EVT_KEY_REPT(KEY_LEFT)) {
```
Comment (bsongis) [inline radio/src/gui/128x64/navigation.cpp]: same problem here

File: radio/src/gui/128x64/navigation.cpp (hunk 5 above)
Comment (kilrah) [reply radio/src/gui/128x64/navigation.cpp]: Well it disables editing with up/down for it too, but we want that for consistency no?

**Comment by kilrah** [comment]
Shall we merge and finish the last things in 2.2?

**Comment by bsongis** [comment]
I have opened a new branch for next changes ...

**Pull request merged**